banking/src/test/java/rocks/kata/AppTest.java

```diff
@@ -0,0 +1,123 @@
+package rocks.kata;
+
+import org.junit.jupiter.api.Test;
+
+import rocks.kata.DatabaseClasses.*;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import org.junit.jupiter.api.BeforeEach;
+
+class AppTest {
+
+    AccountOperationsDatabase database;
+    CurrentDateProvider dateProvider = new CurrentDateProvider();
+    AccountOperationDatabaseFormatter formatter = new AccountOperationDatabaseFormatter();
+
+    @BeforeEach
+    void createAccountDatabase() {
+        database = new AccountOperationsDatabase(formatter);
+    }
+
+    @Test
+    void newlyCreatedAccountShouldResultInCorrectDatabasePrintedStatement() {
+        var account = new Account(database);
+        String currentDate = dateProvider.getCurrentDate();
+
+        String statement = account.printStatement();
+
+        assertEquals("Date\t\tAmount\tBalance\n" + currentDate + "\t0\t0", statement);
+    }
+
+    @Test
+    void accountAfterDepositOperationShouldResultInCorrectDatabasePrintedStatement() {
+        var account = new Account(database);
+        account.deposit(100);
+        String currentDate = dateProvider.getCurrentDate();
+
+        String statement = account.printStatement();
+
+        assertEquals("Date\t\tAmount\tBalance\n" + currentDate + "\t0\t0\n" + currentDate + "\t+100\t100", statement);
+    }
+
+    @Test
+    void depositOperationShouldReturnExceptionWhenAmountIsNegativeNumber() {
+        var account = new Account(database);
+        Exception exception = assertThrows(IllegalArgumentException.class, () -> {
```

Comment:
bloki given-when-then rozjechane są tutaj jakoś, zmniejsza to czytelność - proponuję wyciągnąć lambdę do zmiennej jako `when` i w `then` wszystkie asercje (analogiczna uwaga do pozostałych)